src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposableFieldsShouldBeDisposed.cs

```diff
@@ -77,45 +83,54 @@ void addOrUpdateFieldDisposedValue(IFieldSymbol field, bool disposed)
                 // Instance fields initialized in constructor/method body with a locally created disposable object must be disposed.
                 compilationContext.RegisterOperationBlockStartAction(operationBlockStartContext =>
                 {
-                    if (!(operationBlockStartContext.OwningSymbol is IMethodSymbol containingMethod))
+                    if (!(operationBlockStartContext.OwningSymbol is IMethodSymbol containingMethod) ||
+                        !ShouldAnalyze(containingMethod.ContainingType))
                     {
                         return;
                     }
 
                     if (disposeAnalysisHelper.HasAnyDisposableCreationDescendant(operationBlockStartContext.OperationBlocks, containingMethod))
                     {
-                        if (disposeAnalysisHelper.TryGetOrComputeResult(operationBlockStartContext.OperationBlocks,
-                            containingMethod, operationBlockStartContext.Options, Rule, trackInstanceFields: false,
-                            trackExceptionPaths: false, operationBlockStartContext.CancellationToken,
-                            out var disposeAnalysisResult, out var pointsToAnalysisResult))
+                        PointsToAnalysisResult lazyPointsToAnalysisResult = null;
```

Comment:
>lazyPointsToAnalysisResult [](start = 47, length = 26)

I was gonna ask why cache since we're caching but DFA results, but I guess I should ask, why doesn't DisposeAnalysis cache its results?

Reply:
I turned off caching dispose analysis results as it is unlikely that other analyzers would need to recompute these - we only have 2 dispose analyzers and they operate on different kind of methods (CA2000 on methods that create disposable objects and CA2213 on disposable types with disposable fields and field reference being assigned a disposable object).

Note that is not the case with other common analyses, such as points to analysis, where we know almost certainly that other analyzer(s) will request it.